support RSA256 with PSS padding Signature alg

kalanthales opened the pull request.

ynojima commented on webauthn4j-core/src/main/java/com/webauthn4j/data/SignatureAlgorithm.java:
```diff
@@ -42,6 +43,7 @@ public class SignatureAlgorithm {
     public static final SignatureAlgorithm ES512 = new SignatureAlgorithm(JCA_SHA_512_WITH_ECDSA, SHA512);
     public static final SignatureAlgorithm RS1 = new SignatureAlgorithm(JCA_SHA_1_WITH_RSA, SHA1);
     public static final SignatureAlgorithm RS256 = new SignatureAlgorithm(JCA_SHA_256_WITH_RSA, SHA256);
+    public static final SignatureAlgorithm RS256_PSS = new SignatureAlgorithm(JCA_SHA_256_WITH_RSA_PADDING_PSS, SHA256);
```
To align with COSE name, please rename to PS256.

ynojima reviewed (commented):
PS256 is nice to have. Thank you for pull-request.

- Could you add PS384 and PS512 too?
- Could you add a test case that actually verifies PS256 signature?

By the way, do you know which authenticator or security-key supports PS256?

ynojima commented:
This Pull Request remains in draft mode; do you plan to continue working on it?

kalanthales commented:
Sorry, I'm planning to work on it. 

PS256 is nice to have. Thank you for pull-request.

Could you add PS384 and PS512 too?
Will add
Could you add a test case that actually verifies PS256 signature?
Could you pinpoint me an existing such test case to me. So that I can follow the same convention.
By the way, do you know which authenticator or security-key supports PS256?
Android Key Store attestation supports PS256 and is required by Visa token network key attestation. We plan to use it there.

Thanks.

ynojima commented:
> Could you pinpoint me an existing such test case to me. So that I can follow the same convention.

Could you create an authenticator emulator class that emits PS256 signature by extending [PackedAuthenticator](https://github.com/webauthn4j/webauthn4j/blob/cfceb050fcf05f0bb8610263e4bf746a9c1cf3bc/webauthn4j-test/src/main/java/com/webauthn4j/test/authenticator/webauthn/PackedAuthenticator.java) class and overriding createAttestationStatement method?
Then, create a test case like this: https://github.com/webauthn4j/webauthn4j/blob/955ba688c11b5f97cd707cc8c44b8965813093d9/webauthn4j-core/src/test/java/integration/scenario/webauthn/UserVerifyingAuthenticatorAuthenticationVerificationTest.java#L47-L96

kalanthales commented:
> Is this pull-request ready for review?

Hi @ynojima ,
If you agree with the test I added, I could mark for review. I tried adding a test using packed authenticator. But the base class (WebAuthnModelAuthenticator#makeCredential, WebAuthnModelAuthenticator#isCapableOfHandling) have hard coded EC keys.  So I was reluctant to adding changes in those classes. So if you are ok with the tests I added, could mark as 'Ready for review'. Thanks

ynojima commented:
I made a new pull-request based on your pull-request: https://github.com/webauthn4j/webauthn4j/pull/1107

- Integration test case are added
- Remove bouncy castle provider dependency
-

kalanthales commented:
> I made a new pull-request based on your pull-request: #1107
> 
> * Integration test case are added
> * Remove bouncy castle provider dependency

Thanks  for the new pull-request. Now it looks better and cleaner.

ynojima commented:
Closed as #1107 is merged
